specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+Web Notification APIs
+===
+
+# Background
+
+The WebView2 team is adding support for Web notifications. End developers should
+be able to handle notification permission requests, and further listen to
+`NotificationReceived` events to optionally handle the notifications themselves.
+The `NotificationReceived` events are raised on `CorebWebView2` and
+`CoreWebView2Profile` object respectively for non-persistent and persistent
+notifications respectively.
+
+The `NotificationReceived` event on `CoreWebView2` and `CoreWebView2Profile` let you intercept the web non-persistent and persistent notifications. The host can use the `Notification` property on the `NotificationReceivedEventArgs` to construct an notification matching the look and feel of the host app. The host can also use such information to decide to show or not show a particular notification. The host can `GetDeferral` or set the `Handled` property on the `NotificationReceivedEventArgs` to handle the event at a later time or let WebView2 know if the notification has been handled. By default, if the `NotificationReceived` event is not handled by the host, the web notification will be displayed using the default notification UI provided by WebView2 Runtime.
+
+# Examples
+
+## Handle PermissionRequested event
+
+This can be achieved with the existing `PermissionRequested` events.
+`PermissionRequested` event used to not be raised for `PermissionKind.Notification`.
+Now, `PermissionRequested` events are raised for `PermissionKind.Notification`, and `PermissionState` needs to be set `Allow` explicitly to allow such permission requests. `PermissionState.Default` for `PermissionKind.Notification` is considered denied.
```

Comment:
This needs to be in the ref docs somewhere.

Reply:
Adding it to our ref doc.